services/forgotPassword.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require("jsonwebtoken");
+const User = require("../db/models/User");
+const sendMailService = require("./sendemail");
+const userRepository = require("../repositories/user");
+const env = require("../configuration/env")
+
+async function setResetPasswordToken(user) {
+  const token = jwt.sign({ userId: user._id }, process.env.JWT_SECRET, { expiresIn: '1h' });
+  return token;
+}
+
+async function forgotPassword(email) {
+  const user = await userRepository.getUserByEmail(email);
+
+  if (!user) {
+    throw new Error(errors.EMAIL_NOT_FOUND);
+  }
+
+  const token = await setResetPasswordToken(user);
+
+  const templateName = "forgot-password";
+  const resetPasswordUrl = `${process.env.FRONTEND_URL}/reset-password/`;
```

Comment:
extract "proces.env.FRONTEND_URL" into env.js, so you can reuse it, improves readability, and if you, at any point in development process need to change this, you will need to change in one place instead of changing in every place you use this variable